File: backend/src/main/java/com/project2/domain/member/dto/FollowerResponseDto.java
package com.project2.domain.member.dto;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import com.project2.domain.member.entity.Member;

@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
public class FollowerResponseDto {
    private Long userId;
    private String email;
    private String nickname;
    private String profileImageUrl;
    private int totalPages; // totalPages 추가


    public static FollowerResponseDto fromEntity(Member member) {
        return new FollowerResponseDto(
                member.getId(),
                member.getEmail(),
                member.getNickname(),
                member.getProfileImageUrl(),
                0 // 초기값 설정 (나중에 컨트롤러에서 설정)
        );
    }
}

File: backend/src/main/kotlin/com/project2/domain/member/dto/FollowerResponseDto.kt
package com.project2.domain.member.dto

import com.project2.domain.member.entity.Member

data class FollowerResponseDto(
    val userId: Long, val email: String, val nickname: String, val profileImageUrl: String, val totalPages: Int = 0
) {
    companion object {
        fun from(member: Member): FollowerResponseDto = FollowerResponseDto(

            userId=member.id!!,
            email=member.email,
            nickname=member.nickname,
            profileImageUrl = member.getProfileImageUrlOrDefaultUrl(),
             0
        )
    }
}






